src/main/java/nl/tudelft/jpacman/board/Board.java

```diff
@@ -30,7 +31,7 @@
 	 * Whatever happens, the squares on the board can't be null.
 	 * @return false if any square on the board is null.
 	 */
-	protected final boolean invariant() {
+	protected final boolean invariant(@UnderInitialization(Board.class) Board this) {
```

Comment:
Great! I actually tried this, but did not get it to work and then gave up.

I think I did not give the `Board.class` argument to the `Underinitialization` annotation.